src/test/java/fr/sncf/osrd/infra_state/RouteStateTest.java

```diff
@@ -550,6 +555,60 @@ public void testCircularInfraReserves() {
         }
     }
 
+    @Test
+    public void testTinyInfra2Trains() throws InvalidInfraException {
+        final var config = TestConfig.readResource("tiny_infra/config_railjson_2trains.json");
+        
+        var infra = config.rjsInfra;
+        var switchID = infra.switches.stream().findFirst().get().id;
+        var trainOrderedList = new ArrayList<String>();
+        trainOrderedList.add("First");
+        trainOrderedList.add("Second");
+
+        var initTable = new RJSSuccessionTable(switchID, trainOrderedList.toArray(new String[2]));
+        config.rjsSuccessions = new RJSSuccessions(Collections.singletonList(initTable));
+
+        var simState = config.prepare();
+        simState.run();
+        var sim = simState.sim;
+        var secondTrain = sim.trains.get("Second");
+        var finalPosition = secondTrain.getLastState().location.getPathPosition();
+        var initialPosition = Train.getInitialLocation(secondTrain.schedule, sim).getPathPosition();
+
+        // Check that the second train moves
+        assertNotEquals(initialPosition, finalPosition);
+    }
+
+    @Test
+    public void testChangeSuccessionTable() throws InvalidInfraException {
```

Comment:
The purpose of this test is to change the train order while the simulation is running. The TST is reversed before running the simulation.

Reply:
I don't think this function is usefull anymore.